QuickFIXn/ThreadedSocketAcceptor.cs

```diff
@@ -32,18 +34,56 @@ public class ThreadedSocketAcceptor : IAcceptor
         /// <param name="settings"></param>
         /// <param name="logFactory">If null, a NullFactory will be used.</param>
         /// <param name="messageFactory">If null, a DefaultMessageFactory will be created (using settings parameters)</param>
+        [Obsolete("Use \"Microsoft.Extensions.Logging.ILoggerFactory\" instead of \"QuickFix.Logger.ILogFactory\".")]
         public ThreadedSocketAcceptor(
             IApplication application,
             IMessageStoreFactory storeFactory,
             SessionSettings settings,
             ILogFactory? logFactory = null,
+            IMessageFactory? messageFactory = null) : this(application, storeFactory, settings,
+            logFactory is null ? NullQuickFixLoggerFactory.Instance : new LogFactoryAdapter(logFactory),
+            messageFactory)
+        {
+        }
+
+        /// <summary>
+        /// Create a ThreadedSocketAcceptor
+        /// </summary>
+        /// <param name="application"></param>
+        /// <param name="storeFactory"></param>
+        /// <param name="settings"></param>
+        /// <param name="loggerFactory">If null, a NullFactory will be used.</param>
+        /// <param name="messageFactory">If null, a DefaultMessageFactory will be created (using settings parameters)</param>
+        public ThreadedSocketAcceptor(
+            IApplication application,
+            IMessageStoreFactory storeFactory,
+            SessionSettings settings,
+            ILoggerFactory? loggerFactory = null,
+            IMessageFactory? messageFactory = null) : this(application, storeFactory, settings,
+            loggerFactory is null ? NullQuickFixLoggerFactory.Instance : new MelQuickFixLoggerFactory(loggerFactory),
+            messageFactory)
+        {
+        }
+
+        private ThreadedSocketAcceptor(
+            IApplication application,
+            IMessageStoreFactory storeFactory,
+            SessionSettings settings,
+            IQuickFixLoggerFactory loggerFactory,
             IMessageFactory? messageFactory = null)
         {
-            ILogFactory lf = logFactory ?? new NullLogFactory();
+            if (loggerFactory is LogFactoryAdapter lfa)
+            {
+                // LogFactoryAdapter is only ever created in the constructor marked obsolete, which means we own it and
+                // must save a ref to it so we can dispose it later. Any other loggerFactory is owned by someone else
+                // so we'll leave the dispose up to them. This should be removed eventually together with the old ILog
+                // and ILogFactory.
+                _logFactoryAdapter = lfa;
```

Comment:
@jkulubya Looks like you forgot to add this to `Dispose()`, right?  I'll do it in my current branch.

Reply:
@jkulubya Update: I see you did have a dispose call, but you put it at the end of `Stop()`.  This is an error, because Start/Stop can be called on this object multiple times.  ~Dispose()~ is the right place for it, and I made that change.